server/src/main/scala/endpoints4s/play/server/BasicAuthentication.scala

```diff
@@ -71,19 +71,31 @@ trait BasicAuthentication extends algebra.BasicAuthentication with EndpointsWith
             Valid(tuplerUEHC(tuplerUE(urlData, entityData), tuplerHC(h, credentials)))
         }
       def matchRequest(requestHeader: RequestHeader): Option[RequestEntity[Out]] = {
-        matchRequestAndParseHeaders(requestHeader) {
-          case (_, (_, None /* credentials */ )) =>
-            requestEntityOf(
-              Left(
-                Results.Unauthorized
-                  .withHeaders(HeaderNames.WWW_AUTHENTICATE -> "Basic realm=Realm")
+        if (method.matches(requestHeader)) {
```

Comment:
I am surprised that you don’t also need to match the URL path :thinking:

Reply:
Do you think it would be possible to implement it by still calling `matchRequestAndParseHeaders`?

Reply:
yeah, this is no good, i'll reimplement
can i suggest adding something to testkit that can check if endpoints clash?

like this 
```
  "extra" should {
    "reject unauthenticated requests with Unauthorized response before validating query parameters" in {
      val response = "Hello!"
      import serverApi._
      val protectedEndpointWithParameter = authenticatedEndpoint(
        Get,
        path / "users" / segment[Long]("id"),
        ok(textResponse)
      )
      val unprotectedEndpointWithParameter = endpoint(
        get(path / "users"),
        ok(textResponse)
      )

      serveRoutes(
        routesFromEndpoints(
          protectedEndpointWithParameter.implementedBy(_ => Some(response)),
          unprotectedEndpointWithParameter.implementedBy(_ => response),
        )
      ){ port =>
        val request = HttpRequest(uri = s"http://localhost:$port/users")
        whenReady(sendAndDecodeEntityAsText(request)) { case (_, entity) =>
          entity shouldBe response
          ()
        }
      }
    }
  }
```

Reply:
Yes, any addition to the test kit would be welcome!